exonum/src/storage/proof_map_index/key.rs

```diff
@@ -462,6 +468,53 @@ impl PartialOrd for ProofPath {
     }
 }
 
+impl ProofPath {
+    /// Creates a compressed binary representation of the given proof path.
+    ///
+    /// # Binary format
+    ///
+    /// - **bits_len** - total length in bits compressed by the leb128 algorithm.
+    /// - **bytes** - non-null bytes of the given ProofPath, i.e. the first `(bits_len + 7) / 8` bytes.
+    pub fn compress(&self) -> SmallVec<[u8; 64]> {
+        let bits_len = self.end() as u64;
+        let bytes_len = ((bits_len + 7) / 8) as usize;
+        let key = &self.raw_key()[0..bytes_len];
+
+        let mut buf = smallvec![0u8; 64];
+        let bytes_written = {
+            let mut writer = Cursor::new(buf.as_mut());
+            let mut bytes_written = leb128::write::unsigned(&mut writer, bits_len).unwrap();
+            bytes_written + writer.write(&key).unwrap()
+        };
+        buf.resize(bytes_written, 0);
+        // Cuts of the bits that lie to the right of the end.
+        let last_bits_len = bits_len % 8;
```

Comment:
I suggest smth like this:
```suggestion
        let integer_bytes = (bits_len % 8) == 0;
```

although I don't like `integer_bytes` naming, but `whole_number_of_bytes` is much longer